compiler/modules/types.json/parse.bal

```diff
@@ -49,7 +49,7 @@ function parseType(t:Env env, Binding? b, json j, Path path) returns t:SemType|P
         Typedesc => { return t:TYPEDESC; }
         Handle => { return t:HANDLE; }
         Xml => { return t:XML; } 
-        Json => { return t:createJson(env); }
+        Json => { return t:createJson(t:typeContext(env)); }
```

Comment:
The right fix here is for `parse` to take a `t:Context` rather than a `t:Env`. (Call the parameter `tc`.)